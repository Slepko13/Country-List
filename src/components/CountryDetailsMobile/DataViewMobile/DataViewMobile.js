import React from 'react';
import PropTypes from "prop-types";

import './DataViewMobile.scss';
import vector from '../../../assets/images/Vector 10.png';

const DataViewMobile = (props) => {
    let { title, data, position, addictions, picture } = props;
    data = data === "" ? "n/a" : data === "Kiev" ? "Kyiv" : data;
    return (
        <div className={position ?
            `DataViewMobile ${position}` :
            "DataViewMobile"}
            data-testid="position"
        >
            <img className="ellipse" src={picture} alt='ellipse' />
            <img className="vector" src={vector} alt='vector' />
            <div className="info__title">
                {title}</div>
            <div className="info__content">
                <span style={{ marginLeft: "10px" }}>{addictions}</span>
                {data}
            </div>
        </div>
    )
}
export default DataViewMobile;

DataViewMobile.propTypes = {
    title: PropTypes.string,
    data: PropTypes.string,
    position: PropTypes.string,
    addictions: PropTypes.string,
    picture: PropTypes.string,
}