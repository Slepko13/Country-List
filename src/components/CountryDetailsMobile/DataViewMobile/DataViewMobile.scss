.DataViewMobile {
  margin: 0 0 25px 90px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  position: relative;

  .ellipse {
    position: absolute;
    top: 10px;
    right: 100%;
    margin: 0 25px 0 0;
  }

  .vector {
    position: absolute;
    top: 110%;
    right: 100%;
    margin: 0 25px 0 0;
  }

  .info__title {
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
  }

  .info__content {
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: bold;
    text-align: left;
    color: #313131;
    line-height: 23px;
    font-size: 20px;


  }
}

