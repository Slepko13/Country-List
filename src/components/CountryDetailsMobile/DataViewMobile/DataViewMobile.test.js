import React from "react";
import DataViewMobile from "./DataViewMobile";
import {render} from '@testing-library/react';


describe("my DataViewMobile component", () => {


    it("should render DataViewMobile  Component with props(snapshot)", () => {
        let component = shallow(<DataViewMobile title="Capital" data="London" position="" addictions=""/>);
        expect(component).toMatchSnapshot();
    });

    it("should render DataViewMobile  Component without props(snapshot)", () => {
        let component = shallow(<DataViewMobile/>);
        expect(component).toMatchSnapshot();
    });

    it("should render DataViewMobile  additional classname depending on props ", () => {
        const {getByTestId} = render(
            <DataViewMobile position="left"/>
        );
        expect(getByTestId('position')).toHaveClass('left');
    });

    it("should render DataViewMobile without additional classname depending on props ", () => {
        const {container} = render(
            <DataViewMobile position={null}/>
        );
        expect(container.firstChild.classList.contains('left')).toBe(false)
    });


    it("should render correct data when incoming data is empty string", () => {

        let component = shallow(<DataViewMobile title="Capital" data="" position="" addictions=""/>);
        expect(component.find(".info__content").text()).toEqual('n/a');
    });

    it("should render correct data when incoming data is 'Kiev'", () => {

        let component = shallow(<DataViewMobile title="Capital" data="Kiev" position="" addictions=""/>);
        expect(component.find(".info__content").text()).toEqual('Kyiv');
    });

    it("should render correct data with custom incoming data", () => {

        let component = shallow(<DataViewMobile title="Capital" data="Paris" position="" addictions=""/>);
        expect(component.find(".info__content").text()).toEqual('Paris');
    });

});

