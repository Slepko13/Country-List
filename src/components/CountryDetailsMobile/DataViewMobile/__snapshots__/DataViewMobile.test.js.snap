// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my DataViewMobile component should render DataViewMobile  Component with props(snapshot) 1`] = `
<div
  className="DataViewMobile"
  data-testid="position"
>
  <img
    alt="ellipse"
    className="ellipse"
  />
  <img
    alt="vector"
    className="vector"
    src="Vector 10.png"
  />
  <div
    className="info__title"
  >
    Capital
  </div>
  <div
    className="info__content"
  >
    London
    <span
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    />
  </div>
</div>
`;

exports[`my DataViewMobile component should render DataViewMobile  Component without props(snapshot) 1`] = `
<div
  className="DataViewMobile"
  data-testid="position"
>
  <img
    alt="ellipse"
    className="ellipse"
  />
  <img
    alt="vector"
    className="vector"
    src="Vector 10.png"
  />
  <div
    className="info__title"
  />
  <div
    className="info__content"
  >
    <span
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    />
  </div>
</div>
`;
